src/sort.rs

```diff
@@ -126,14 +187,31 @@ fn sort_array(arr: &mut Array) {
     arr.set_trailing_comma(trailing_comma);
 }
 
-fn sort_table(table: &mut Table, group: bool) {
+fn sort_table(table: &mut Table, group: bool, target_tables: &TargetTablePaths) {
     if group {
         sort_by_group(table);
+    } else if !target_tables.is_empty() {
+        // The `table` name must be `target`
+        for (_key, paths) in target_tables {
+            for path in paths {
+                if path.len() > 1 {
+                    sort_table_by_path(table, &path[1..]);
+                }
+            }
+        }
     } else {
         table.sort_values();
     }
 }
 
+fn sort_table_by_path(table: &mut Table, path: &[String]) {
+    if path.is_empty() {
+        table.sort_values();
+    } else if let Some(Item::Table(inner_table)) = table.get_mut(&path[0]) {
+        sort_table_by_path(inner_table, &path[1..]);
+    }
```

Comment:
```suggestion
    let Some(first) = path.get(0) else {
        table.sort_values();
        return;
    };
    
    if let Some(Item::Table(inner_table)) = table.get_mut(&first) {
        sort_table_by_path(inner_table, &path[1..]);
    }
```

This ties together the invariant of `path.is_empty` with the indexing into it.

Reply:
Done